ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -259,7 +299,7 @@ void disable() {
      * @return {@code true} if the scheduling was successful. {@code false} if not.
      */
     boolean scheduleNextForCompaction(BlobStore store) {
-      if (!enabled || !store.isStarted() || storesToSkip.contains(store)) {
+      if (!enabled || !adminBasedTriggerEnabled || !store.isStarted() || storesToSkip.contains(store)) {
```

Comment:
Perhaps the error message in `AmbryRequests` could now include "trigger compaction failed, check if adminBasedTrigger is enabled" or some such comment?